pkg/cmd/server/api/types.go

```diff
@@ -46,8 +46,8 @@ type NodeConfig struct {
 	// DNSIP holds the IP
 	DNSIP string
 
-	// NetworkPluginName is a string specifying the networking plugin
-	NetworkPluginName string
+	// NodeNetworkConfig provides network plugin information
+	NodeNetworkConfig NodeNetworkConfig
```

Comment:
or you could add the nested struct in v1, keep the old field, name it "DeprecatedNetworkPluginName", set omitempty, and move the value to the struct in defaulting